packages/vscode/src/plugins/prisma-postgres-manager/ConnectionStringStorage.ts

```diff
@@ -0,0 +1,120 @@
+import type { SecretStorage } from 'vscode'
+import z from 'zod'
+
+// Storing connection strings in this JSON structure under a single key in the secret storage.
+// This allows for easy deletion of multiple entries when a project is deleted or the user logs out from a workspace.
+const StoredConnectionStringsSchema = z.object({
+  workspaces: z.record(
+    z.string(), // workspaceId
+    z.object({
+      projects: z.record(
+        z.string(), // projectId
+        z.object({
+          databases: z.record(
+            z.string(), // databaseId
+            z.object({
+              connectionString: z.string(),
+            }),
+          ),
+        }),
+      ),
+    }),
+  ),
+})
+
+type StoredConnectionStrings = z.infer<typeof StoredConnectionStringsSchema>
+
+const STORED_CREDENTIALS_KEY = 'prisma-postgres.connection-strings'
+
+export class ConnectionStringStorage {
+  constructor(private readonly vscodeSecretStorage: SecretStorage) {}
+
+  async storeConnectionString({
+    workspaceId,
+    projectId,
+    databaseId,
+    connectionString,
+  }: {
+    workspaceId: string
+    projectId: string
+    databaseId: string
+    connectionString: string
+  }): Promise<void> {
+    const storage = await this.getStoredConnectionStrings()
+
+    const updatedStorage: StoredConnectionStrings = {
```

Comment:
Do we actually need such deeply nested structure? Aren't database IDs unique themselves independently of the workspaces and projects?

Reply:
If we only store the databases by id we'd have to keep another structure to know which database belongs to which project and workspace so we can wipe all credentials for a specific project or workspace upon deletion/logout.

Am not 100% sure about id collisions. So far I always treat them as namespaces to be on the safe side.

We could work with a list here as well and filter that list then. 🤔 

like:
```
[
  {
    workspaceId: '123',
    projectId: '456',  
    databaseId: '789',
  },
  ...
]
```